spring-boot/proxy/src/test/java/LocalApiKeyAndServiceApiManagerTest.java

```diff
@@ -0,0 +1,116 @@
+/*-
+ * #%L
+ * coatrack-proxy
+ * %%
+ * Copyright (C) 2013 - 2021 Corizon | Institut für angewandte Systemtechnik Bremen GmbH (ATB)
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+import eu.coatrack.api.ApiKey;
+
+import eu.coatrack.proxy.security.AdminCommunicator;
+import eu.coatrack.proxy.security.LocalApiKeyAndServiceApiManager;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import static org.junit.jupiter.api.Assertions.*;
+
+import java.sql.Timestamp;
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+
+public class LocalApiKeyAndServiceApiManagerTest extends LocalApiKeyAndServiceApiManager{
+
+    private final String someValidApiKeyValue = "someValidApiKeyValue";
+
+    private ApiKey apiKey;
+
+    private final long
+            oneHourInMillis = 1000 * 60 * 60,
+            oneDayInMillis = oneHourInMillis * 24;
+
+    private final Timestamp
+            now = new Timestamp(System.currentTimeMillis()),
+            tomorrow = new Timestamp(now.getTime() + oneDayInMillis),
+            yesterday = new Timestamp(now.getTime() - oneDayInMillis),
+            halfAnHourAgo = new Timestamp(now.getTime() - oneHourInMillis / 2);
+
+    public LocalApiKeyAndServiceApiManagerTest() {
+        super(null);
+    }
+
+    @BeforeEach
+    public void createAnAcceptingDefaultSetup(){
+        buildUpApiKey();
+        buildUpVerifier();
+    }
+
+    private void buildUpApiKey() {
+        apiKey = new ApiKey();
+        apiKey.setKeyValue(someValidApiKeyValue);
+        apiKey.setDeletedWhen(null);
+        apiKey.setValidUntil(tomorrow);
+    }
+
+    private void buildUpVerifier() {
+        localApiKeyList = new ArrayList<>();
+        localApiKeyList.add(apiKey);
+        latestLocalApiKeyListUpdate = LocalDateTime.now();
+    }
+
+    @Test
+    public void isDefaultKeyAccepted(){
+        assertTrue(isApiKeyValidConsideringTheLocalApiKeyList(apiKey.getKeyValue()));
+    }
+
+    @Test
+    public void testIfDeletedKeyIsDenied(){
```

Comment:
Maybe rename the tests so that the name indicates the expected behaviour? In this case: `deletedApiKeyShouldBeDenied`

This could also be applied to the other tests below.

Reply:
Yes, that is more readable. Done.